### PR TITLE
Use the current indentation as reference when pasting

### PR DESCRIPTION
**Pull request by tgodzik**
Previously, we would try to guess the current indentation for the user, but since we don't know exactly what is their intention, it was easy guess wrong. Now we use the current indentation to see what user intends.

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/formatting/IndentOnPaste.scala [hunk 1]
```diff
@@ -171,17 +153,7 @@ object IndentOnPaste extends RangeFormatter {
...
-    // drop leading/trailing empty lines
-    val lastIdx = idented.length - 1
-    val range = 0 to lastIdx
-    val trimmedStart =
-      range.dropWhile(converted(_).isEmpty).headOption.getOrElse(0)
-    val trimmedEnd = range.reverse
-      .dropWhile(converted(_).isEmpty)
-      .headOption
-      .getOrElse(lastIdx) + 1
-    idented.slice(trimmedStart, trimmedEnd)
```
Comment (dos65) [inline metals/src/main/scala/scala/meta/internal/metals/formatting/IndentOnPaste.scala:184]: There were some issues with this?

**Review by dos65 — APPROVED** [review]
Looks good! Now it's just an indentation corrector without any non-reliable logic.

File: metals/src/main/scala/scala/meta/internal/metals/formatting/IndentOnPaste.scala (hunk 1 above)
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/formatting/IndentOnPaste.scala:184]: There is no need to trim the pasted snippet, it's also unexpected behaviour.

File: metals/src/main/scala/scala/meta/internal/metals/formatting/IndentOnPaste.scala [hunk 2]
```diff
@@ -77,20 +71,14 @@ object IndentOnPaste extends RangeFormatter {
...
-    val prePastedLines = splitLines.take(startLine).reverse
+    val pastedLines = splitLines.slice(startLine, endLine + 1)
 
     // Do not adjust indentation if we pasted into an existing line content
     def pastedIntoNonEmptyLine = {
       pastedLines match {
-        case Array(singleLine) =>
-          val originalLine = singleLine.substring(0, originalStart) +
-            singleLine.substring(endPos.endColumn)
+        case array if array.length > 0 =>
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/formatting/IndentOnPaste.scala:79]: I changed the logic here to not reformat if we are printing into any line that already contains code.

File: tests/unit/src/test/scala/tests/rangeFormatting/MultilineStringRangeFormattingWhenSelectingSuite.scala [hunk 3]
```diff
@@ -110,23 +110,23 @@ class MultilineStringRangeFormattingWhenSelectingSuite
     s"""
        |object Main {
```
Comment (tgodzik) [inline tests/unit/src/test/scala/tests/rangeFormatting/MultilineStringRangeFormattingWhenSelectingSuite.scala:111]: The change here is due to the new way of calculating expected indent, I think we should at some point detect whether we are pasting or selecting, but not sure exactly how to do that.

File: metals/src/main/scala/scala/meta/internal/metals/formatting/IndentOnPaste.scala (hunk 1 above)
Comment (dos65) [reply metals/src/main/scala/scala/meta/internal/metals/formatting/IndentOnPaste.scala:184]: Ok. I don't remember exactly what was the reason. Probably, I added this just to have less changes in test data :smile:

**Review by dos65 — APPROVED** [review]
:rocket:

**Pull request merged**